### PR TITLE
[Named Actor]throw RayException when getting named actor timed out

### PR DESCRIPTION
**Pull request by WangTaoTheTonic**
## Why are these changes needed?
Before getting named actor with Java we didn't separate exception caused by non-existing and timed out, but both return an `Optional.empty()`, which confuses users.
We can throw an exception if timed out when getting named actor, so that users know they should retry or something.
***This will lead to API change.*

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by jovany-wang** [comment]
I don't think this is an API change, because even before this PR, we don't guarantee it returns null if timed out either. So no need to wait to release this until the time of next major version.

And is it better to use a specified exception(RayTimeoutException) instead of RayException? Because there may throw other runtime exceptions which is hard to be distinguished for user programming.

**Comment by WangTaoTheTonic** [comment]
I'd rather not to add a new type of Exception as `RayException` now means runtime exception in Ray cluster, user should retry when it's thrown. There's is not need for user to sense the reason is timed out or some other system errors.

**Review by jovany-wang — APPROVED** [review]
LGTM overall, excepting 2 minor comments.

File: java/test/src/main/java/io/ray/test/NamedActorTest.java [hunk 1]
```diff
@@ -98,4 +98,8 @@ public void testActorDuplicatedName() {
     // Registering with the same name should fail.
     Ray.actor(Counter::new).setName(name).remote();
   }
+
+  public void test_getNonExistingNamedActor() {
```
[the remark this answers is not in the document]
Comment (WangTaoTheTonic) [reply java/test/src/main/java/io/ray/test/NamedActorTest.java:102]: nice catch!

File: src/ray/core_worker/lib/java/io_ray_runtime_RayNativeRuntime.cc [hunk 2]
```diff
@@ -292,15 +292,15 @@ Java_io_ray_runtime_RayNativeRuntime_nativeGetActorIdOfNamedActor(JNIEnv *env, j
...
+  const auto pair = CoreWorkerProcess::GetCoreWorker().GetNamedActorHandle(
+      full_name, /*ray_namespace=*/"");
+  const auto status = pair.second;
+  if (status.IsNotFound()) {
+    return IdToJavaByteArray<ActorID>(env, ActorID::Nil());
   }
+  THROW_EXCEPTION_AND_RETURN_IF_NOT_OK(env, status, nullptr);
+  const auto actor_handle = pair.first;
+  ActorID actor_id = actor_handle->GetActorID();
   return IdToJavaByteArray<ActorID>(env, actor_id);
```
[the remark this answers is not in the document]
Comment (WangTaoTheTonic) [reply src/ray/core_worker/lib/java/io_ray_runtime_RayNativeRuntime.cc:304]: nice catch two!

File: java/test/src/main/java/io/ray/test/NamedActorTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jovany-wang) [reply java/test/src/main/java/io/ray/test/NamedActorTest.java]: > Does this format of method name is recommended in somewhere?

**Comment by jovany-wang** [comment]
Windows failure unrelated.

**Pull request merged**